yew-macro/src/stringify.rs

```diff
@@ -1,68 +1,91 @@
 use proc_macro2::TokenStream;
 use quote::{quote, quote_spanned, ToTokens};
 use syn::spanned::Spanned;
-use syn::{Expr, Lit};
+use syn::{Expr, Ident, Lit, LitStr};
 
-/// Attempt converting expression to str, if it's a literal
-pub fn try_stringify_expr(src: &Expr) -> Option<String> {
-    match src {
-        Expr::Lit(l) => try_stringify_lit(&l.lit),
-        _ => None,
+/// Stringify a dynamic value at runtime.
+pub fn stringify_dynamic(src: impl ToTokens) -> TokenStream {
+    quote_spanned! {src.span()=>
+        ::std::borrow::Cow::<'static, str>::Owned(
+            ::std::string::ToString::to_string(&(#src)),
+        )
     }
 }
 
-/// Attempt converting literal to str literal
-fn try_stringify_lit(src: &Lit) -> Option<String> {
-    match src {
-        Lit::Str(v) => Some(v.value()),
-        Lit::Char(v) => Some(v.value().to_string()),
-        Lit::Int(v) => Some(v.base10_digits().to_string()),
-        Lit::Float(v) => Some(v.base10_digits().to_string()),
-        Lit::Bool(v) => Some(v.value.to_string()),
-        _ => None,
+/// Map an `Option` type such that it turns into `Cow<'static, str>`.
+pub fn stringify_option_at_runtime(src: impl ToTokens) -> TokenStream {
+    let ident = Ident::new("__yew_str", src.span());
+    let sr = stringify_dynamic(&ident);
+    quote! {
+        ::std::option::Option::map(#src, |#ident| {
+            #sr
+        })
     }
 }
 
-/// Converts literals and expressions to Cow<'static, str> construction calls
-pub struct Constructor(TokenStream);
-
-macro_rules! stringify_at_runtime {
-    ($src:expr) => {{
-        let src = $src;
-        Self(quote_spanned! {src.span()=>
-            ::std::borrow::Cow::<'static, str>::Owned(
-                ::std::string::ToString::to_string(&#src),
-            )
-        })
-    }};
+/// Create `Cow<'static, str>` construction calls.
+///
+/// This is deliberately not implemented for strings to preserve spans.
```

Comment:
When you use a string literal in `quote!` it doesn't have a `Span` attached to it so if there's an error involving it the span will point to the entire call site. The `Stringify` trait isn't implemented for any `String` type to avoid that kind of mistake

Reply:
I see, thanks for the explanation!